File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.cpp
class Solution {
   public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        int n = nums.size();
        vector<int> freq(n - 2);
        vector<int> ans;
        for (int num : nums) {
            if (freq[num] == 1) ans.emplace_back(num);
            freq[num]++;
        }

        return ans;
    }
};

File: 3581-the-two-sneaky-numbers-of-digitville/3581-the-two-sneaky-numbers-of-digitville.rs
impl Solution {
    pub fn get_sneaky_numbers(nums: Vec<i32>) -> Vec<i32> {
        let mut ans = Vec::new();
        let mut freq = vec![0; nums.len() - 2];

        for num in nums {
            if freq[num as usize] == 1 {
                ans.push(num);
            }
            freq[num as usize] += 1;
        }

        ans
    }
}